### PR TITLE
Add String and SecureString to acceptable credential types in PSCredentialInfo

### LINKED ISSUE #652: PSResourceRepository -CredentialInfo should support all SM secret types
### Summary of the new feature / enhancement

PSResourceRepository -CredentialInfo should support all SM secret types so that a PSCredential Object is not required. 
PowerShell SecretManagement supports the following secret data types:

- byte[]
- string
- SecureString
- PSCredential
- Hashtable

### Proposed technical implementation details (optional)

_No response_

### PR DESCRIPTION
**Pull request by alerickson**
# PR Summary
Currently `Register-PSResourceRepository` and `Set-PSResourceRepository` both have a `-CredentialInfo` parameter, which takes a type of `PSCredentialInfo`.  This type includes the vault name, secret name, and `PSCredential` object, all used to configure credential persistence with SecretManagement.  

This PR allows for credentials of type `String` and `SecureString` to be passed in as a `Credential` within `PSCredentialInfo`

## PR Context

Resolves #652 

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Documentation needed]()
        - [ ] Issue filed: 
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**
    - [x] [Documentation needed]()
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary.

### CODE REVIEW

**Review by anamnavi — APPROVED** [review]
LGTM

File: src/code/PSCredentialInfo.cs [hunk 1]
```diff
@@ -43,7 +45,20 @@ public PSCredentialInfo(PSObject psObject)
 
             VaultName = (string) psObject.Properties[PSCredentialInfo.VaultNameAttribute]?.Value;
             SecretName = (string) psObject.Properties[PSCredentialInfo.SecretNameAttribute]?.Value;
-            Credential = (PSCredential) psObject.Properties[PSCredentialInfo.CredentialAttribute]?.Value;
+
+            var credentialAttr = psObject.Properties[PSCredentialInfo.CredentialAttribute]?.Value;
+            if (credentialAttr is String)
```
Comment (PaulHigin) [inline src/code/PSCredentialInfo.cs]:
```suggestion
            if (credentialAttr is string credStr)
```

File: src/code/PSCredentialInfo.cs [hunk 2]
```diff
@@ -43,7 +45,20 @@ public PSCredentialInfo(PSObject psObject)
...
+            {
+                Credential = new PSCredential("PSGetUser", new NetworkCredential("", (String) credentialAttr).SecurePassword);
```
Comment (PaulHigin) [inline src/code/PSCredentialInfo.cs]:
```suggestion
                Credential = new PSCredential("PSGetUser", new NetworkCredential("", credStr).SecurePassword);
```

File: src/code/PSCredentialInfo.cs [hunk 3]
```diff
@@ -43,7 +45,20 @@ public PSCredentialInfo(PSObject psObject)
...
+            }
+            else if ((credentialAttr as PSObject)?.BaseObject is SecureString)
```
Comment (PaulHigin) [inline src/code/PSCredentialInfo.cs]:
```suggestion
            else if ((credentialAttr as PSObject)?.BaseObject is SecureString credSS)
```

File: src/code/PSCredentialInfo.cs [hunk 4]
```diff
@@ -43,7 +45,20 @@ public PSCredentialInfo(PSObject psObject)
...
+            {
+                Credential = new PSCredential("PSGetUser", (credentialAttr as PSObject).BaseObject as SecureString);
```
Comment (PaulHigin) [inline src/code/PSCredentialInfo.cs]:
```suggestion
                Credential = new PSCredential("PSGetUser", credSS);
```

File: src/code/PSCredentialInfo.cs [hunk 5]
```diff
@@ -43,7 +45,20 @@ public PSCredentialInfo(PSObject psObject)
...
+            }
+            else if (credentialAttr is PSCredential)
```
Comment (PaulHigin) [inline src/code/PSCredentialInfo.cs]:
```suggestion
            else if (credentialAttr is PSCredential psCred)
```

File: src/code/PSCredentialInfo.cs [hunk 6]
```diff
@@ -43,7 +45,20 @@ public PSCredentialInfo(PSObject psObject)
...
+            {
+                Credential = (PSCredential)credentialAttr;
```
Comment (PaulHigin) [inline src/code/PSCredentialInfo.cs]:
```suggestion
                Credential = psCred;
```

**Review by PaulHigin — CHANGES_REQUESTED** [review]

**Review by alyssa1303 — COMMENTED** [review]
LGTM as well

**Review by alyssa1303 — APPROVED** [review]

**Review by PaulHigin — APPROVED** [review]
LGTM

**Pull request merged**